elephant_vending_machine/libraries/vending_machine.py

```diff
@@ -0,0 +1,192 @@
+"""Abstract GPIO pin setup and interactions
+
+This module contains funcitonality required to create
+a vending machine instance, with nested SensorGrouping
+instances used as an abstraction of the different screens
+and associated sensors on the vending machines.
+
+.. todo::
+
+   Finish implementing
+   Determine pin numbers
+"""
+
+import multiprocessing as mp
+import time
+from gpiozero.pins.pigpio import PiGPIOFactory
+from gpiozero import MotionSensor, LED
+import spur
+
+
+LEFT_SCREEN = 1
+MIDDLE_SCREEN = 2
+RIGHT_SCREEN = 3
+MOTION_PIN = 10
+LED_PIN = 3
+RED = 33
+GREEN = 34
+
+
+def worker(i, addresses):
+    """ The function which is used by the pool. It calls the wait_for_detection() method
+        for the passed associated SensorGrouping and returns the value returned by that method.
+    """
+    group = None
+    if i == LEFT_SCREEN:
+        group = SensorGrouping(addresses[0], LEFT_SCREEN)
+    elif i == MIDDLE_SCREEN:
+        group = SensorGrouping(addresses[1], MIDDLE_SCREEN)
+    else:
+        group = SensorGrouping(addresses[2], RIGHT_SCREEN)
+    result = group.wait_for_detection()
+    return result
+
+
+class VendingMachine:
+    """Provides an abstraction of the physical 'vending machine'.
+
+    This class provides an abstraction of the sensors and screen associated
+    with the elephant vending machine.
+    """
+
+    def __init__(self, addresses, config):
+        """Initialize an instance of VendingMachine.
+
+        Provides an interface to the different sensors in divided by their associated screen.
+
+        Parameters:
+            addresses (list): A list of local IP addresses of the Raspberry Pis
+            config (dict): The configuration values of the Flask server
+        """
```

Comment:
Good point, will ask in the next meeting

Reply:
Up to you if you want to do this. I think it might be good to consider, but also in the interest of time it may be safe to assume she's not going to be adding screens.

Reply:
I'm gonna make the assumption that the physical changes that would have to be made to the machine would be too much.